src/parse.rs

```diff
@@ -321,6 +321,69 @@ pub fn parse_mailbox(
     }
 }
 
+pub fn parse_status(
```

Comment:
We really should write a macro to help write these `parse_*` methods @mordak. They all have basically the same outer structure. Doesn't need to happen as part of this PR though.